packages/theme/pages/MyAccount.vue

```diff
@@ -62,56 +47,21 @@ export default {
     OrderHistory,
     OrderDetails
   },
-  middleware: ['is-authenticated'],
+  middleware: ['is-authenticated', 'my-profile'],
   setup(props, context) {
-    const route = useRoute();
     const router = useRouter();
     const { logout } = useUser();
-    const isMobile = computed(mapMobileObserver().isMobile);
-
-    const handleOpenPage = async ({ pageName }) => {
-      const pageRoute = {
-        ...route.value,
-        path: context.root.localePath(route.value.path),
-        params: { pageName }
-      };
-      router.push(pageRoute);
-    };
 
     const handleLogout = async () =>
       logout()
         .then(() => context.root.localePath({ name: 'home' }))
         .then(path => router.push(path));
 
-    const pages = computed(() => [
-      { pageName: 'my-profile', i18nKey: 'pages.my_account.content_page_title_my_profile', onActive: handleOpenPage },
-      { pageName: 'saved-addresses', i18nKey: 'pages.my_account.content_page_title_saved_addresses', onActive: handleOpenPage },
-      { pageName: 'order-history', i18nKey: 'pages.my_account.content_page_title_order_history', onActive: handleOpenPage },
-      { pageName: 'order-details', i18nKey: 'pages.my_account.content_page_title_order_details' },
-      { pageName: 'log-out', i18nKey: 'pages.my_account.content_page_title_log_out', onActive: handleLogout }
-    ].map((page) => ({ ...page, localizedTitle: context.root.$i18n.t(page.i18nKey) })));
-
-    const findPageByPageName = (name) => pages.value.find(({ pageName }) => pageName === name);
-
-    const findPageByLocalizedTitle = (title) => pages.value.find(({ localizedTitle }) => localizedTitle === title);
-
-    const changeActivePage = async (title) => {
-      const page = findPageByLocalizedTitle(title);
-      if (!page) return handleOpenPage({ pageName: undefined });
-      await page?.onActive?.(page);
-    };
-
-    const activePage = computed(() => {
-      const { pageName } = route.value.params;
-      if (!pageName) return isMobile.value ? {} : findPageByPageName('my-profile');
-      return findPageByPageName(pageName);
-    });
-
     onBeforeUnmount(() => {
       unMapMobileObserver();
```

Comment:
The mobileObserver is no longer in use. The onBeforeUnmount lifecycle hook and unMapMobileObserver import should both be removed.